controllers/users-controller.mjs

```diff
@@ -0,0 +1,143 @@
+import jsSHA from 'jssha';
+
+const SALT = process.env.SALT_PASSWORD;
+
+const getHashSalted = (input) => {
+  // create new SHA object
+  const shaObj = new jsSHA('SHA-512', 'TEXT', { encoding: 'UTF8' });
+  // create an unhashed cookie string based on user ID and salt
+  const unhashedString = `${input}-${SALT}`;
+
+  // generate a hashed cookie string using SHA object
+  shaObj.update(unhashedString);
+
+  return shaObj.getHash('HEX');
+};
+
+export default function initUsersController(db) {
+  const signup = async (req, res) => {
+    console.log(req.body);
+    const {
+      firstName, lastName, password, email, username,
+    } = req.body;
+
+    try {
+      const user = await db.User.findOne({
+        where: {
+          email,
+        },
+      });
+      console.log('user', user);
+
+      if (!user) {
+        const hashedPassword = getHashSalted(password);
+        console.log('hashed password', hashedPassword);
+
+        const newUser = {
+          firstName,
+          lastName,
+          username,
+          email,
+          password: hashedPassword,
+        };
+        console.log(newUser);
+
+        const userNew = await db.User.create(newUser);
+
+        res.send({
+          email: userNew.email,
+        });
+      } else {
+        res.status(409).send({
+          error: 'The email address is already in use.',
+        });
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const login = async (req, res) => {
+    const { email, password } = req.body;
+
+    try {
+      const user = await db.User.findOne({
+        where: {
+          email,
+        },
+      });
+      console.log('user found!');
+      const hashedPassword = getHashSalted(password);
+      if (hashedPassword === user.password) {
+        const unhashedCookieString = `${user.id}`;
+        const hashedCookieString = getHashSalted(unhashedCookieString);
+        res.cookie('loggedInHash', hashedCookieString);
+
+        const loggedInUser = {
+          id: user.id,
+          email: user.email,
+          firstName: user.firstName,
+          lastName: user.lastName,
+          username: user.username,
+        };
+        res.cookie('user', JSON.stringify(loggedInUser));
+
+        const result = {
+          user: loggedInUser,
+        };
+
+        res.send(result);
```

Comment:
Can we use `res.json` for responses 
following https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/
```suggestion
        res.status(200).json({result: result, message: user successfully login});
```

Reply:
Learnt sth new. Will do, thanks!